### ISSUE TITLE
Training Loss Graphs for training_nli.py

### ISSUE THREAD

**Issue by nandinib1999** [UKPLab/sentence-transformers#423]
Hi,

If I would like to plot the training_loss graph for training_nli.py, is this the correct approach?

```
        running_loss = 0.0
        tstep = 0
        for epoch in trange(epochs_trained, epochs, desc="Epoch"):
            training_steps = 0
            if resume:
                training_steps = steps_trained_in_current_epoch
                resume = False

            for loss_model in loss_models:
                loss_model.zero_grad()
                loss_model.train()

            print("training_steps ", training_steps)
            print("steps_per_epoch ", steps_per_epoch)
            for _ in trange(training_steps, steps_per_epoch, desc="Iteration", smoothing=0.05):
                start_time = time.time()
                for train_idx in range(num_train_objectives):
                    loss_model = loss_models[train_idx]
                    optimizer = optimizers[train_idx]
                    scheduler = schedulers[train_idx]
                    data_iterator = data_iterators[train_idx]

                    try:
                        data = next(data_iterator)
                    except StopIteration:
                        #logging.info("Restart data_iterator")
                        data_iterator = iter(dataloaders[train_idx])
                        data_iterators[train_idx] = data_iterator
                        data = next(data_iterator)

                    features, labels = batch_to_device(data, self._target_device)
                    loss_value = loss_model(features, labels)
                    running_loss += loss_value.item()
                    if fp16:
                        with amp.scale_loss(loss_value, optimizer) as scaled_loss:
                            scaled_loss.backward()
                        torch.nn.utils.clip_grad_norm_(amp.master_params(optimizer), max_grad_norm)
                    else:
                        loss_value.backward()
                        torch.nn.utils.clip_grad_norm_(loss_model.parameters(), max_grad_norm)

                    optimizer.step()
                    scheduler.step()
                    optimizer.zero_grad()
                    
                training_steps += 1
                global_step += 1

                if evaluation_steps > 0 and ((training_steps - start_eval_step) % evaluation_steps == 0 or training_steps == start_eval_step):
                     values = self._eval_during_training(evaluator, output_path, save_best_model, epoch, global_step, eval_type, optimizer, scheduler)
                     for loss_model in loss_models:
                          loss_model.zero_grad()
                          loss_model.train()

                    esteps_list.append(global_step)
                    print("Time ", time.time() - start_time)
                    print("Loss ", running_loss / evaluation_steps)
                    if len(train_loss_list) <= 2000:
                        train_loss_list.append(running_loss / evaluation_steps)
                    else:
                        train_loss_list = []
                        train_loss_list.append(running_loss / evaluation_steps)
                    plt.figure(1)
                    plt.plot(esteps_list, train_loss_list, color='green')
                    plt.title('Training Loss')
                    plt.xlabel('Global Steps')
                    plt.ylabel('Training Loss')
                    plt.savefig('plot-train-loss.png')
                    running_loss = 0.0
```
Do you have the training loss graphs that I can look at and verify my plots?

**Comment by nreimers** [comment]
I never plotted any graph, so I cannot help here.
